widget/embedded/src/components/SameTokensWarning/SameTokensWarning.tsx

```diff
@@ -0,0 +1,30 @@
+import { i18n } from '@lingui/core';
+import { Alert, Divider, NoRouteIcon, Typography } from '@rango-dev/ui';
+import React from 'react';
+
+import { useQuoteStore } from '../../store/quote';
+import { areTokensEqual } from '../../utils/wallets';
+
+import { Container } from './SameTokensWarning.styles';
+
+export function SameTokensWarning() {
+  const { fromToken, toToken } = useQuoteStore();
+  const showWarningMessage =
+    fromToken && toToken && areTokensEqual(fromToken, toToken);
```

Comment:
The type of `showWarningMessage` is inferred as `boolean | null`. Using `!!fromToken` and `!!toToken` would ensure the inferred type is strictly `boolean`, making it better aligned with the variable's name.

Reply:
Fixed